app/src/test/java/com/volchok/rocketapp/feature/details/presentation/DetailsViewModelTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.volchok.rocketapp.feature.details.presentation
+
+import com.volchok.rocketapp.feature.details.domain.OpenRocketLaunchUseCase
+import com.volchok.rocketapp.library.api.model.details.RocketDetailsModel
+import com.volchok.rocketapp.library.data.model.Data
+import com.volchok.rocketapp.library.rockets.domain.FetchRocketInfoUseCase
+import com.volchok.rocketapp.library.rockets.domain.ObserveRocketDetailsUseCase
+import com.volchok.rocketapp.library.use_case.domain.invoke
+import io.kotest.matchers.shouldBe
+import io.mockk.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.*
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.*
+
+internal class DetailsViewModelTest {
+
+    private val fetchRocketInfoUseCase = mockk<FetchRocketInfoUseCase>()
+    private val observeRocketDetailsUseCase = mockk<ObserveRocketDetailsUseCase>()
+    private val openRocketLaunchUseCase = mockk<OpenRocketLaunchUseCase>()
+
+    private val testRocketDetailsModel = mockk<RocketDetailsModel>()
+
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Before
+    fun setUp() {
+        val dispatcher = UnconfinedTestDispatcher()
+        Dispatchers.setMain(dispatcher)
+    }
+
+    @Test
+    fun `if opens rocket launch screen`() = runTest {
+
+        coEvery { fetchRocketInfoUseCase.invoke() } returns Data.Success(testRocketDetailsModel)
+
+        coEvery { observeRocketDetailsUseCase.invoke() } returns flowOf(testRocketDetailsModel)
+
+        every { openRocketLaunchUseCase.invoke(Unit) } just runs
+
+        val detailsViewModel = DetailsViewModel(
```

Comment:
v jednotlivych testech to pouzijes a usetri to par radek kodu

`val detailsViewModel = createViewModel()`

Reply:
Opraveno